project_forecast_line_priority/models/res_config_settings.py

```diff
@@ -0,0 +1,87 @@
+# Copyright 2024 Therp BV
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+SELECTION_FORECAST_ENDDATE = [
+    ("none", "None"),
+    ("date", "Date"),
+    ("delta", "Delta (in days)"),
+]
+
+
+class ResConfigSettings(models.TransientModel):
+    _inherit = "res.config.settings"
+
+    forecast_line_priority_0_date = fields.Date(
```

Comment:
I had to think hard about the double date fields (one as string, one as Date) for each priority. I understand that in ir.config.parameter you only can have string values. But if the date fields where added to the company (meaning you could have different settings per company as well, although not relevant for our own internal purposes), and as related field in res.config.settings, this would not be a problem at all. Just as we do with default journals in account-financial-tools/account_move_default_journal.

Reply:
I have also a question about the use case for this. Suppose we say all high priority tasks muse be done by the 1st of march 2024. Obviously these dates will need updating all the time. When will this be done and by whom?

Another thing, should the update of these fields not be linked to the task stage in some way? For instance only be set when going out of draft status and be frozen when done or cancelled?

Reply:
> I had to think hard about the double date fields (one as string, one as Date) for each priority. I understand that in ir.config.parameter you only can have string values. But if the date fields where added to the company (meaning you could have different settings per company as well, although not relevant for our own internal purposes), and as related field in res.config.settings, this would not be a problem at all. Just as we do with default journals in account-financial-tools/account_move_default_journal.

I agree about this, I'd like to hear @gjotten  opinion too

Reply:
> Another thing, should the update of these fields not be linked to the task stage in some way? For instance only be set when going out of draft status and be frozen when done or cancelled?

A useful shortcut to avoid triggering forecast computation, and therefore ignoring the planned end date altogether, is writing `forecast_role_id = False`  on project.task records where `state in ['pending', 'cancelled', 'done']` and writing `forecast_role_id = 1` where `state in ['draft','open']`.

It's not pretty, but spread over across two base.automation records in our own db it does the trick: we only consider tasks that have an active stage's state. Of course this is still only possible because we have a single forecast role defined, but we avoid having to do anything based on task status in our module(s) through this.

Now that I thought on it a bit more, maybe we run the risk of having the task priority calculation for forecasting becoming non-performant for a lot of records though if not manually configured correctly. On the other hand, as far as I understand it, the big implementation leading to the creation of `project_forecast_line` by Camptocamp was set up to have these kinds of duct-taped solutions as well. It's just a not-so-fleshed out module and we're improving things step-by-step.
 
> I agree about this, I'd like to hear @gjotten opinion too

I don't _love_ yet another company setting, but I think I agree it makes sense when wanting to use a cleaner solution than the date to string one. Per-company settings I hadn't even taken into account...

@NL66278 with an eye out for OCA implementation, would you say we _have_ to take this res.company route?

Reply:
@gjotten Have to is a big word, but I think it will certainly help adoption. Single company instances are not bothered with it at all, and for multi company instances it might be crucial.

Reply:
:+1: okay then let's proceed with it; @ntsirintanis over to you